cli/cmd/import/import.go

```diff
@@ -0,0 +1,137 @@
+	"github.com/agntcy/dir/importer/types"
+	"github.com/spf13/cobra"
+)
+
+var (
+	cfg          config.Config
+	registryType string
+	factory      *types.Factory
+)
+
+var Command = &cobra.Command{
+	Use:   "import",
+	Short: "Import records from external registries",
+	Long: `Import records from external registries into DIR.
+
+Supported registries:
+  - mcp: Model Context Protocol registry v0.1
+
+The import command fetches records from the specified registry and pushes
+them to DIR.
+
+Examples:
+  # Import from MCP registry
+  dirctl import --type=mcp --url=https://registry.modelcontextprotocol.io
+
+  # Import with filters
+  # Available filters: https://registry.modelcontextprotocol.io/docs#/operations/list-servers-v0.1#Query-Parameters
+  dirctl import --type=mcp --url=https://registry.modelcontextprotocol.io --filter=updated_since=2025-08-07T13:15:04.280Z
+
+  # Preview without importing
+  dirctl import --type=mcp --url=https://registry.modelcontextprotocol.io --dry-run
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return runImport(cmd)
+	},
+}
+
+func init() {
+	// Initialize factory and register importers once at startup
+	factory = types.NewFactory()
+	mcp.Register(factory)
+
+	// Add flags
+	Command.Flags().StringVar(&registryType, "type", "", "Registry type (mcp, a2a)")
+	Command.Flags().StringVar(&cfg.RegistryURL, "url", "", "Registry base URL")
+	Command.Flags().StringToStringVar(&cfg.Filters, "filter", nil, "Filters (key=value)")
+	Command.Flags().IntVar(&cfg.Limit, "limit", 0, "Maximum number of records to import (0 = no limit)")
+	Command.Flags().BoolVar(&cfg.DryRun, "dry-run", false, "Preview without importing")
+
+	// Mark required flags
+	Command.MarkFlagRequired("type") //nolint:errcheck
+	Command.MarkFlagRequired("url")  //nolint:errcheck
+}
+
+func runImport(cmd *cobra.Command) error {
+	// Get the client from the context
+	c, ok := ctxUtils.GetClientFromContext(cmd.Context())
+	if !ok {
+		return errors.New("failed to get client from context")
+	}
+
+	// Set the registry type from the string flag
+	cfg.RegistryType = config.RegistryType(registryType)
+
+	// Set the client
+	cfg.Client = c
+
+	// Validate configuration
+	if err := cfg.Validate(); err != nil {
+		return fmt.Errorf("invalid configuration: %w", err)
+	}
+
+	// Create importer instance from pre-initialized factory
+	importer, err := factory.Create(cfg)
+	if err != nil {
+		return fmt.Errorf("failed to create importer: %w", err)
+	}
+
+	// Run import with progress reporting
+	presenter.Printf(cmd, "Starting import from %s registry at %s...\n", cfg.RegistryType, cfg.RegistryURL)
+
+	if cfg.DryRun {
+		presenter.Printf(cmd, "Mode: DRY RUN (preview only)\n")
+	}
+
+	presenter.Printf(cmd, "\n")
+
+	result, err := importer.Run(cmd.Context(), cfg)
+	if err != nil {
+		return fmt.Errorf("import failed: %w", err)
+	}
+
+	// Print summary
+	printSummary(cmd, result)
+
+	return nil
+}
+
+func printSummary(cmd *cobra.Command, result *types.ImportResult) {
```

Comment:
i think the import summary should be split in two separate summaries:
- import stats summary - collects useful stats about full imports
- import summary - collects each item summary, e.g. `{item, status}` where we can collect all items based on success. we can do this in a separate issue/PR to be able to easily support retry mechanisms

Both summaries can be handled as JSON, for example

- `dirctl import --type=mcp --url=https://registry.modelcontextprotocol.io` - prints stats summary RAW by default
- `dirctl import --type=mcp --url=https://registry.modelcontextprotocol.io --json` - prints stats as JSON
- `dirctl import --type=mcp --url=https://registry.modelcontextprotocol.io --status status.json` - stores import summary as JSON for each item

Reply:
Let's address this in a separate PR, yes. Regarding the import summary, it would be good to discuss what would be the use case where this is needed before implementing. The sync feature does not provide us this type of report either.